Zend/zend_API.h

```diff
@@ -1972,6 +2016,30 @@ static zend_always_inline int zend_parse_arg_class_name_or_obj(
 	return 1;
 }
 
+static zend_always_inline int zend_parse_arg_str_or_object(
+	zval *arg, zend_string **destination_string, zval **destination_object, zend_class_entry *base_ce, int allow_null
```

Comment:
I'm not sure what type the object destination should be.

Reply:
I personally would use a zend_object as I find the current behaviour of the object ZPP a bit weird that it returns a zval, but that's only me.

Reply:
I've changed it to `zend_object` :)